drivers/i2c/i2c_ll_stm32.c

```diff
@@ -435,6 +471,9 @@ static int i2c_stm32_pm_action(const struct device *dev, enum pm_device_action a
 	switch (action) {
 	case PM_DEVICE_ACTION_RESUME:
 		err = i2c_stm32_activate(dev);
+        #if defined(CONFIG_SOC_SERIES_STM32WLX)
+        i2c_stm32_reinit_timing(dev);
```

Comment:
Why not calling already available `i2c_stm32_runtime_configure` ?
Why not calling it from `i2c_stm32_activate` ?

Reply:
- Found that was no necessary to repeat all operation from `i2c_stm32_runtime_configure `
- `i2c_stm32_activate ` Is called else where, my thinking was that timing reinit wasn't needed at every call of `i2c_stm32_activate `.

Do you think it's still relevant to create the `i2c_stm32_reinit_timing` function or just call `i2c_stm32_runtime_configure ` inside `i2c_stm32_activate ` is enough ?

Reply:
> Do you think it's still relevant to create the i2c_stm32_reinit_timing function or just call i2c_stm32_runtime_configure  inside i2c_stm32_activate  is enough ?

I see 2 benefits:
- smaller code increase
- validated code reuse (as for instance your proposal would have missed the clock_off call)

So, if my proposal is functional, I'd suggest to make use of it.

This being said, I recall that correctly reinitializing I2C when existing Stop2 used to require significant changes such as https://github.com/zephyrproject-rtos/zephyr/pull/60998. See https://github.com/zephyrproject-rtos/zephyr/issues/59194#issuecomment-1988808489.
Btw, maybe @adrienbruant you'd be able to confirm if current PR is sufficient (which may be the case since several changes have been introduced in PM subsystem since that time) ?

Reply:
`i2c_stm32_runtime_configure` is a relatively big function, that at the end just write one `uint32_t` value to a I2C register using `LL_I2C_SetTiming`. An alternative would be to just store this value in `i2c_stm32_data`, and always write the register before the transaction, within the block protected from suspending the device or going to stop modes.

That would also avoid configuring registers in `i2c_stm32_runtime_configure` that anyway are going to be lost when the device goes into stop 2.

On the drawback, it's a much bigger code change, so maybe it's an optimization to be kept for later.

Reply:
> That would also avoid configuring registers in i2c_stm32_runtime_configure that anyway are going to be lost when the device goes into stop 2.

Note that this is specific to WL

> `i2c_stm32_runtime_configure` is a relatively big function, that at the end just write one `uint32_t` value to a I2C register using `LL_I2C_SetTiming`. An alternative would be to just store this value in `i2c_stm32_data`, and always write the register before the transaction, within the block protected from suspending the device or going to stop modes.

This is a fair point. I'm fine with this proposal as well.

Reply:
In our branch, we use the PM_NOTIFIER introduced by #60998 to restore from STOP2. Our restore routine is not as well thought out as we call i2c_stm32_configure_clk_and_registers(), a looong function.

@erwango Can you point to the changes introduced in the PM subsystem? This might work for us, in the sense that the peripheral is now restored from the pessimist assumption that it needs to be fully reconfigured. Isn't that a suboptimal solution for STOP1 and lighter sleep modes?

Reply:
> i2c_stm32_runtime_configure is a relatively big function, that at the end just write one uint32_t value to a I2C register using LL_I2C_SetTiming. An alternative would be to just store this value in i2c_stm32_data,

@aurel32  Can you point me out how to do so ?

As I understand changes  include: 
- Create new `uint32_t i2c_clock` var inside `i2c_stm32_data`
- Store the value at initialisation using `dev->data.i2c_clock = i2c_clock`
- Use `LL_I2C_SetTiming(dev->data.i2c_clock)` inside `i2c_stm32_activate`

Reply:
> In our branch, we use the PM_NOTIFIER introduced by #60998 to restore from STOP2. Our restore routine is not as well thought out as we call i2c_stm32_configure_clk_and_registers(), a looong function.
> 
> @erwango Can you point to the changes introduced in the PM subsystem? This might work for us, in the sense that the peripheral is now restored from the pessimist assumption that it needs to be fully reconfigured. Isn't that a suboptimal solution for STOP1 and lighter sleep modes?

Yes, this is suboptimal, but this is a first step before mitigations can be applied:
- make the reconfiguration less costly as suggested above
- add PM runtime support (similar to #72991 on the SPI side) to disable the device as soon as it is not needed anymore, that way there is no need to save it around a STOP transition. As a bonus is reduces the runtime power consumption.

Reply:
> @erwango Can you point to the changes introduced in the PM subsystem? This might work for us, in the sense that the peripheral is now restored from the pessimist assumption that it needs to be fully reconfigured. Isn't that a suboptimal solution for STOP1 and lighter sleep modes?

@adrienbruant I was referring to : 
https://github.com/zephyrproject-rtos/zephyr/pull/70623
https://github.com/zephyrproject-rtos/zephyr/pull/70111

Now, note that whole PM subsystem is not yet fully mature and this is likely to evolve again. See https://github.com/zephyrproject-rtos/zephyr/issues/75963